### ISSUE TITLE
Stack Overflow Error in DMN Parser

### ISSUE THREAD

**Issue by pihme** [camunda/zeebe#9545]
**Describe the bug**
Observed in production logs: https://console.cloud.google.com/errors/detail/CIHn_-jg7uShyQE;service=zeebe;time=P7D?project=camunda-cloud-240911

in cluster: 4ae19f04-89da-40e4-9021-2b77029acf1a

**Expected behavior**

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
 java.lang.StackOverflowError: null
	at org.camunda.bpm.model.xml.impl.util.DomUtil.filterNodeList(DomUtil.java:140) ~[camunda-xml-model-7.16.0.jar:7.16.0]
	at org.camunda.bpm.model.xml.impl.util.DomUtil.filterNodeListByType(DomUtil.java:184) ~[camunda-xml-model-7.16.0.jar:7.16.0]
	at org.camunda.bpm.model.xml.impl.instance.DomElementImpl.getChildElementsByType(DomElementImpl.java:136) ~[camunda-xml-model-7.16.0.jar:7.16.0]
	at org.camunda.bpm.model.xml.impl.type.child.ChildElementCollectionImpl.getView(ChildElementCollectionImpl.java:82) ~[camunda-xml-model-7.16.0.jar:7.16.0]
	at org.camunda.bpm.model.xml.impl.type.child.ChildElementCollectionImpl.access$000(ChildElementCollectionImpl.java:38) ~[camunda-xml-model-7.16.0.jar:7.16.0]
	at org.camunda.bpm.model.xml.impl.type.child.ChildElementCollectionImpl$1.iterator(ChildElementCollectionImpl.java:164) ~[camunda-xml-model-7.16.0.jar:7.16.0]
	at scala.collection.convert.JavaCollectionWrappers$JCollectionWrapper.iterator(JavaCollectionWrappers.scala:72) ~[scala-library-2.13.8.jar:?]
	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:98) ~[scala-library-2.13.8.jar:?]
	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:87) ~[scala-library-2.13.8.jar:?]
	at scala.collection.convert.JavaCollectionWrappers$JCollectionWrapper.map(JavaCollectionWrappers.scala:68) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.parseDecision(DmnParser.scala:118) ~[dmn-engine-1.7.1.jar:1.7.1]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$3(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:454) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$2(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:100) ~[scala-library-2.13.8.jar:?]
	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:87) ~[scala-library-2.13.8.jar:?]
	at scala.collection.mutable.ArrayBuffer.map(ArrayBuffer.scala:43) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.parseDecision(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$3(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:454) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$2(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:100) ~[scala-library-2.13.8.jar:?]
	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:87) ~[scala-library-2.13.8.jar:?]
	at scala.collection.mutable.ArrayBuffer.map(ArrayBuffer.scala:43) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.parseDecision(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$3(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:454) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$2(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:100) ~[scala-library-2.13.8.jar:?]
	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:87) ~[scala-library-2.13.8.jar:?]
...
```

</p>
</details>

**Environment:**
- OS: Camunda Cloud
- Zeebe Version: 8.0.2
- Configuration:

**Comment by pihme** [comment]
Cluster is at 100% backpressure and both partitions are marked as unhealthy. 
It is a trial cluster so it doesn't have too much resources anyway.

**Comment by pihme** [comment]
Related question in slack: https://camunda.slack.com/archives/CHAC0L80M/p1655445272755469

**Comment by saig0** [comment]
I can reproduce the issue with the following DMN:

![image](https://user-images.githubusercontent.com/4305769/174292254-0bedfe36-75d2-4730-8006-b8841fc3eb4f.png)

The DMN contains a loop. "Decision 1" requires "Decision 2" and "Decision 2" requires "Decision 1". 

At the moment, the DMN doesn't detect loops (see the todo [here](https://github.com/camunda-community-hub/dmn-scala/blob/master/dmn-engine/src/main/scala/org/camunda/dmn/parser/DmnParser.scala#L115-L120)).

**Comment by saig0** [comment]
@remcowesterhoud yes. We should detect loops during the parsing of the DMN. As a result, the deployment should be rejected. 

Please create an issue in the DMN engine to add the check.

This is also a good candidate for the (upcoming) linter of the modeler.

**Comment by korthout** [comment]
Sev high because it can break a cluster. We should prioritize working on the dmn-scala issue and backport the dependency update.

**Comment by oleschoenburg** [comment]
We are getting more and more reports of this now, with slightly different stack traces:

```
 java.lang.StackOverflowError: null
	at scala.collection.JavaConverters$.collectionAsScalaIterableConverter(JavaConverters.scala:261) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.parseDecision(DmnParser.scala:116) ~[dmn-engine-1.7.1.jar:1.7.1]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$3(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:454) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$2(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:100) ~[scala-library-2.13.8.jar:?]
	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:87) ~[scala-library-2.13.8.jar:?]
	at scala.collection.mutable.ArrayBuffer.map(ArrayBuffer.scala:43) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.parseDecision(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$3(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:454) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$2(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:100) ~[scala-library-2.13.8.jar:?]
	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:87) ~[scala-library-2.13.8.jar:?]
	at scala.collection.mutable.ArrayBuffer.map(ArrayBuffer.scala:43) ~[scala-library-2.13.8.jar:?]
	at org.camunda.dmn.parser.DmnParser.parseDecision(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
	at org.camunda.dmn.parser.DmnParser.$anonfun$parseDecision$3(DmnParser.scala:120) ~[dmn-engine-1.7.1.jar:1.7.1]
```

here it's `DmnParser.scala:116` and `DmnParser.scala:120` whereas above it was `DmnParser.scala:118` and `DmnParser.scala:120`

I've linked them to this issue here anyway, let me know if that's okay.

**Comment by menski** [comment]
fixed with the release of DMN scala 1.7.3 release
